[REF] Update the graph reset docs to reflect volumes (#198)
neurobagel/documentation: docs/maintaining.md

```

!!! note "Specify the deployment profile"

    Whenever you use `docker compose` commands, you also have to specify
    the deployment profile you want to use with the `-p` or `--profile` flag.
    If you forget this step, `docker compose` will use the default profile (`local_node`),
    and will only shut down, update, or start the services for that profile.

## Updating the data in your graph

If you have followed the [initial setup](getting_started.md)
and have deployed your Neurobagel node from our Docker Compose recipe,

> your node will have a dedicated graph database that only contains 
> the data for your node.

Reviewer: ```suggestion
your node now has a dedicated graph database that stores
the datasets for your node.
By default, the graph database will only contain an [example dataset called `BIDS synthetic`](https://github.com/neurobagel/recipes/blob/main/data/example_synthetic_pheno-bids.jsonld). 
```

> This makes updating the data in your graph a straightforward process.

Reviewer: ```suggestion
Replacing the existing data in your graph database with your own data (or updated data) is a straightforward process.
```

> Once you have generated the updated files you want to upload,

Reviewer: ```suggestion
Once you have generated or updated the JSONLD files you want to upload,
```

the process to update the data in your graph is:

> 1. Shut down the Neurobagel node

Reviewer: same comment as below - the numbered points don't seem to be showing up correctly in the preview 🤔

    ```bash
    docker compose --profile full_stack down

>     ```

Reviewer: ```suggestion
    ```

   (or, replace `full_stack` with the profile you are using)
```

> 2. Update the data files in [your `LOCAL_GRAPH_DATA` directory](config.md#uploading-data-to-the-graph-store)

Reviewer: I find it kind of confusing to have to navigate to a section under the "Manual setup" steps just to see how to change the default toy data, and in this case I think it's also not necessary. How about just directly stating what they need to change in the .env file:

```suggestion
2. In your `.env` file, replace the value of the `LOCAL_GRAPH_DATA` variable with the path to the directory containing your JSONLD files.
```

I think we can further expand on this section a bit in:
- https://github.com/neurobagel/documentation/issues/202

so that we do not really need to refer to the section `config.md#uploading-data-to-the-graph-store` (and just have it for reference, as with the other manual setup steps)

Reviewer: What do you think about applying the au

3. Restart the Neurobagel node

    ```bash
    docker compose --profile full_stack up -d
    ```

> Here are some common scenarios where you might need to update the data in your graph:

Reviewer: ```suggestion
Here are some other common scenarios where you might need to update the data in your graph:
```

    Whenever you use `docker compose` commands, you also have to specify
    the deployment profile you want to use with the `-p` or `--profile` flag.
    If you forget this step, `docker compose` will use the default profile (`local_node`),
    and will only shut down, update, or start the services for that profile.

## Updating the data in your graph

If you have followed the [initial setup](getting_started.md)
and have deployed your Neurobagel node from our Docker Compose recipe,
your node now has a dedicated graph database that stores
the datasets for your node.
By default, the graph database will only contain an [example dataset called `BIDS synthetic`](https://github.com/neurobagel/recipes/blob/main/data/example_synthetic_pheno-bids.jsonld). 

Replacing the existing data in your graph database with your own data (or updated data) is a straightforward process.
Once you have generated or updated the JSONLD files you want to upload,
the process to update the data in your graph is:

1. Shut down the Neurobagel node

    ```bash
    docker compose --profile full_stack down
    ```

   (or, replace `full_stack` with the profile you are using)

> 2. Update the data files in [your `LOCAL_GRAPH_DATA` directory](config.md#uploading-data-to-the-graph-store)

Reviewer: ```suggestion
2. Update the data files in the directory specified by the [`LOCAL_GRAPH_DATA` environment variable](config.md#uploading-data-to-the-graph-store), or simply change the path to the directory containing your JSONLD files.
```

### Resetting your GraphDB instance

Each Neurobagel node has its own GraphDB instance, which is used to store the graph data for the node.
If you want to reset your graph database and start again from scratch,
follow these steps:

> 1. Ensure that your Neurobagel node is not running (i.e., shut down the Docker containers for the node).

Reviewer: In the docs preview I don't see these numbered points rendered correctly - could you double check this?

Author: Interesting, that seems to be true of all the numbered lists here. Not sure why

Author: seems related, but apparently this was fixed a long time ago: https://github.com/squidfunk/mkdocs-material/discussions/3294

Author: OK, tailwind was the culprit here. In fact, our production docs don't have any visible ordered list elements right now. Kinda spooky - maybe we need some visual integration tests ...

    ```bash
    docker compose --profile full_stack down
    ```

    If you are not using the `full_stack` profile, replace `full_stack` with the name of the profile you are using.

> 2. Delete the docker volume that contains the GraphDB data for your node.

Reviewer: ```suggestion
2. Delete the docker volume that contains the GraphDB data for your node.

!!! warning "This will wipe all GraphDB user credentials and datasets in the node"
This step will clear all existing GraphDB configuration settings, ensuring that the next time you [launch the node](getting_started.md#launch-neurobagel) (which will automatically configure GraphDB for you again), you are starting from a fresh GraphDB instance without risks of conflicts with existing credentials or databases.
```

Author: I see what you mean, but this warning adds more confusion than it resolves in my view. "This will wipe all GraphDB user credentials and datasets in the node" sounds like a horrible thing that you would like to avoid. But unless I somehow take over the graph container after launch to start configuring things by hand this will never be a problem. The whole thing is supposed to be ephemeral. 

If you think this needs more context, let's add it to the list of "general docs refactor" in a section of "what other things might I configure in graphDB, how, and what to watch out for (i.e. wiping it)"

    ```bash

>     docker volume rm neurobagel_graphdb_data

Reviewer: To ensure we don't run into volume name clashes or repetition so that it is easy to identify which volume is associated with which deployment (e.g., recipes_graphdb_data, recipes_graphdb_data-1 gets confusing fast), here's an issue to document the `COMPOSE_PROFILE_NAME`:

- See https://github.com/neurobagel/documentation/issues/201

Reviewer: ```suggestion
    docker volume rm neurobagel_node_graphdb_home
```
same as above for graphdb_home, and `neurobagel_node` should be the default `COMPOSE_PROFILE_NAME`

    ```

>     replace `neurobagel_graphdb_data` with the name of the volume that was created for your node.

Reviewer: ```suggestion
    replace `neurobagel_node_graphdb_home` with the name of the volume that was created for your node.
```
same as above

    It is usually named `<project_name>_graphdb_data`

>     where `<project_name>` is the name of the directory where you have stored the `docker-compose.yml` file for your Neurobagel node.

Reviewer: How about a little info/tip admonition here that a user can double check the known volumes with

```bash
docker volume ls
```
?

Could save them from having to recheck their .env file if they forget the name of the volume

Reviewer: also, in this case b/c the `template.env` specifies a default `COMPOSE_PROJECT_NAME`, I think the project name for now is always some variant on `neurobagel_node` rather than the project directory (which should always be `recipes`)

3. Start your Neurobagel node again.

    ```bash
    docker compose --profile full_stack up -d

>     ```

Reviewer: ```suggestion
    ```

    If you are not using the `full_stack` profile, replace `full_stack` with the name of the profile you are using.
```

Some examples of when you might want to do this:

- You started but did not complete Neurobagel node setup previously and want to ensure you are using up-to-date instructions and recommended configuration options
- Your local node has stopped working after a configuration change to your graph database (e.g., your Neurobagel node API no longer starts or responds with an error, but you have confirmed all environment variables you have set should be correct)

> !!! warning

Reviewer: given the suggested info box above, I think we can remove this! (probably not super useful to be at the very end anyways)

### Resetting your GraphDB instance

Each Neurobagel node has its own GraphDB instance, which is used to store the graph data for the node.
If you want to reset your graph database and start again from scratch,
follow these steps:

1. Ensure that your Neurobagel node is not running (i.e., shut down the Docker containers for the node).

    ```bash
    docker compose --profile full_stack down
    ```

    If you are not using the `full_stack` profile, replace `full_stack` with the name of the profile you are using.

2. Delete the docker volume that contains the GraphDB data for your node.

    ```bash
    docker volume rm neurobagel_node_graphdb_home
    ```

    replace `neurobagel_node_graphdb_home` with the name of the volume that was created for your node.

>     It is usually named `<project_name>_graphdb_data`

Reviewer: ```suggestion
    It is usually named `<project_name>_graphdb_home`
```
